### PR TITLE
modifty generic_data_parser to save assay_tree_rule if it is a new pr…

### PR DESCRIPTION
**Pull request by goshiro2010**
…otocol. Enabled save of 'experiment details' if it is present in the file. Save SEL column order and track if column is marked as a condition. SEL columna attributes are used by downstream query apps such as ACAS dataViewer.

### CODE REVIEW

File: modules/GenericDataParser/src/server/generic_data_parser.R [hunk 1]
```diff
@@ -2822,12 +2916,14 @@ runMain <- function(pathToGenericDataFormatExcelFile, reportFilePath=NULL,
     } else {
       calculatedResults$experimentID <- experiment$id
       calculatedResults$experimentVersion <- experiment$version
+      saveSession('/tmp/beforeUploadData.rda')
```
Comment (bffrost) [inline modules/GenericDataParser/src/server/generic_data_parser.R]: Should this still be in here?

File: modules/GenericDataParser/src/server/generic_data_parser.R [hunk 2]
```diff
@@ -2677,6 +2752,17 @@ runMain <- function(pathToGenericDataFormatExcelFile, reportFilePath=NULL,
   if (!file.exists(fullPathToFile)) {
     stop("SEL: Cannot find input file")
   }
+
+## TEST ###
```
Comment (bffrost) [inline modules/GenericDataParser/src/server/generic_data_parser.R]: This stuff looks like it should be removed.

File: modules/GenericDataParser/src/server/generic_data_parser.R [hunk 3]
```diff
@@ -1898,6 +1972,7 @@ createNewExperiment <- function(metaData, protocol, lsTransaction, pathToGeneric
                                  lsTags=lsTags)
   
   # Save the experiment to the server
+  #saveSession('/tmp/experiment.rda')
```
Comment (bffrost) [inline modules/GenericDataParser/src/server/generic_data_parser.R]: Can we remove this line too?

File: modules/GenericDataParser/src/server/generic_data_parser.R (hunk 1 above)
Comment (brianbolt) [inline modules/GenericDataParser/src/server/generic_data_parser.R]: should remove this

File: modules/GenericDataParser/src/server/generic_data_parser.R [hunk 4]
```diff
@@ -1214,10 +1250,15 @@ organizeCalculatedResults <- function(calculatedResults, inputFormat, formatPara
...
   
   # Call the function that extracts valueKinds, units, conc, concunits from the headers
-  valueKinds <- extractValueKinds(calculatedResultsValueKindRow, ignoreTheseAsValueKinds, uncertaintyType, 
+  valueKinds <- extractValueKinds(valueKindsVector = calculatedResultsValueKindRow, ignoreHeaders = ignoreTheseAsValueKinds, uncertaintyType = uncertaintyType, 
                                   uncertaintyCodeWord, commentCol, commentCodeWord, stateAssignments, 
                                   hiddenColumns, linkColumns, classRow, stateKindRow, stateTypeRow, 
-                                  codeAssignments = codeAssignments)
+                                  codeAssignments = codeAssignments, conditionColumns)
+
+#(valueKindsVector, ignoreHeaders = NULL, uncertaintyType, uncertaintyCodeWord,
```
Comment (bffrost) [inline modules/GenericDataParser/src/server/generic_data_parser.R]: Any reason to have this commented out code still in here?

File: modules/GenericDataParser/src/server/generic_data_parser.R (hunk 3 above)
Comment (brianbolt) [inline modules/GenericDataParser/src/server/generic_data_parser.R]: not really needed comment

File: modules/GenericDataParser/src/server/generic_data_parser.R [hunk 5]
```diff
@@ -2677,6 +2752,17 @@ runMain <- function(pathToGenericDataFormatExcelFile, reportFilePath=NULL,
...
+
+## TEST ###
+.libPaths('/opt/acas-dataexplorer/acas/r_libs')
+library(racas)
+library('RCurl')
+source('/opt/acas-dataexplorer/acas/src/r/GenericDataParser/generic_data_parser.R')
+fullPathToFile <- '/opt/acas-dataexplorer/acas/privateUploads/Tier1c_SEL_test3 (21).xlsx'
+recordedBy <- 'goshiro'
+configList <- racas::applicationSettings
+lsTransaction <- 1234
```
Comment (brianbolt) [inline modules/GenericDataParser/src/server/generic_data_parser.R]: should remove most of this

**Comment by goshiro2010** [comment]
Nope. I will remove the saveSession calls

On Jul 28, 2017, at 8:53 AM, bffrost <notifications@github.com> wrote:

*@bffrost* requested changes on this pull request.

Looks like some testing / debugging lines need to be removed before this is
production-ready.
------------------------------

In modules/GenericDataParser/src/server/generic_data_parser.R
<https://github.com/mcneilco/acas/pull/193#discussion_r130123590>:

> @@ -2822,12 +2916,14 @@ runMain <- function(pathToGenericDataFormatExcelFile, reportFilePath=NULL,
     } else {
       calculatedResults$experimentID <- experiment$id
       calculatedResults$experimentVersion <- experiment$version
+      saveSession('/tmp/beforeUploadData.rda')

Should this still be in here?
------------------------------

In modules/GenericDataParser/src/server/generic_data_parser.R
<https://github.com/mcneilco/acas/pull/193#discussion_r130123714>:

> @@ -2677,6 +2752,17 @@ runMain <- function(pathToGenericDataFormatExcelFile, reportFilePath=NULL,
   if (!file.exists(fullPathToFile)) {
     stop("SEL: Cannot find input file")
   }
+
+## TEST ###

This stuff looks like it should be removed.
------------------------------

In modules/GenericDataParser/src/server/generic_data_parser.R
<https://github.com/mcneilco/acas/pull/193#discussion_r130123775>:

> @@ -1898,6 +1972,7 @@ createNewExperiment <- function(metaData, protocol, lsTransaction, pathToGeneric
                                  lsTags=lsTags)

   # Save the experiment to the server
+  #saveSession('/tmp/experiment.rda')

Can we remove this line too?
------------------------------

In modules/GenericDataParser/src/server/generic_data_parser.R
<https://github.com/mcneilco/acas/pull/193#discussion_r130123927>:

>                                    uncertaintyCodeWord, commentCol, commentCodeWord, stateAssignments,
                                   hiddenColumns, linkColumns,
classRow, stateKindRow, stateTypeRow,
-                                  codeAssignments = codeAssignments)
+                                  codeAssignments = codeAssignments,
conditionColumns)
+
+#(valueKindsVector, ignoreHeaders = NULL, uncertaintyType,
uncertaintyCodeWord,

Any reason to have this commented out code still in here?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/mcneilco/acas/pull/193#pullrequestreview-52968665>, or mute
the thread

File: modules/GenericDataParser/src/server/generic_data_parser.R [hunk 6]
```diff
@@ -3079,7 +3177,7 @@ organizeSubjectData <- function(
                                             lockCorpBatchId= F, errorEnv= errorEnv, precise = precise, link = link, 
                                             calculateGroupingID = preciseTreatmentGroupID, 
                                             stateAssignments = stateAssignments, concColumn = concColumn,
-                                            codeAssignments = codeAssignments)
+                                            codeAssignments = codeAssignments)[[1]]
```
Comment (brianbolt) [inline modules/GenericDataParser/src/server/generic_data_parser.R]: might be safer to not assume it's the first one in this list but rather specific select the subject data, but up to you

File: modules/GenericDataParser/src/server/generic_data_parser.R [hunk 7]
```diff
@@ -3393,3 +3491,118 @@ external.parseGenericData <- function(request) {
...
+                                    lsKind = "condition column",
+                                    lsTransaction = lsTransaction,
+                                    recordedBy = recordedBy,
+                                    stringValue = as.character(dataRow$conditionColumn)
+      )}
+#	return(removeEmptyValues_DV(values))
+	return(values)
+}
+createColumnOrderStates <- function(exptDataColumns=selColumnOrderInfo, errorEnv, recordedBy, lsTransaction){
+exptDataColumns=selColumnOrderInfo
```
Comment (brianbolt) [inline modules/GenericDataParser/src/server/generic_data_parser.R]: probably should assign with <- and add 2 spaces to the beginning of the line

File: modules/GenericDataParser/src/server/generic_data_parser.R (hunk 6 above)
Comment (goshiro2010) [inline modules/GenericDataParser/src/server/generic_data_parser.R]: Good point. I will leave it as-is for now.

**Pull request merged**